packages/microcosm/src/addons/model.js

```diff
@@ -22,108 +23,140 @@ function invoke(binding: *, repo: Microcosm, scope: Object): * {
   return binding
 }
 
+type Answer = { [string]: * }
+
 export default class Model extends Emitter {
-  repo: Microcosm
-  scope: Object
-  bindings: { [key: string]: * }
-  subscriptions: { [key: string]: * }
-  revision: number
-  value: *
+  _scope: Object
+  _repo: Microcosm
+  _bindings: *[]
+  _subscriptions: { [string]: * }
+  _pendingUpdate: ?Job
+  _publish: () => void
+  _patch: Answer
+
+  value: Answer
 
   constructor(repo: Microcosm, scope: Object) {
     super()
 
-    this.repo = repo
-    this.scope = scope
-    this.bindings = {}
-    this.subscriptions = {}
+    this._repo = repo
+    this._scope = scope
+    this._bindings = []
+    this._subscriptions = {}
+    this._patch = {}
+    this._publish = this._publish.bind(this)
+    this._pendingUpdate = null
+
     this.value = {}
-    this.revision = 0
+  }
+
+  bind(bindings: Object) {
+    this._bindings.length = 0
+
+    for (var key in bindings) {
+      let callback = bindings[key]
+
+      if (isObservable(callback)) {
+        this._track(key, callback)
+      } else {
+        this._bindings.push({ key, callback })
+      }
+    }
 
-    this.repo.on('change', this.compute, this)
+    if (this._bindings.length) {
+      this._repo.on('change', this._compute, this)
+    } else {
+      this._repo.off('change', this._compute, this)
+    }
```

Comment:
If we're only using Observables in the model, then don't listen to the Microcosm instance at all. Instead rely on those Observables to "push" in new updates.

Reply:
Hmmm. This is going to attach a new listener without cleaning up the old one. Fixing.